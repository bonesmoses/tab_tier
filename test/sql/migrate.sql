\set ECHO none

BEGIN;
\set ECHO all

CREATE EXTENSION tab_tier;

CREATE SCHEMA tiertest;

SET search_path TO tiertest;

CREATE TABLE tier_test (foo INT, dt TIMESTAMP WITH TIME ZONE);

INSERT INTO tier_test
SELECT a.id, '2016-12-15'::DATE - (a.id::TEXT || 'd')::INTERVAL
  FROM generate_series(1, 200) a (id);

SELECT tab_tier.register_tier_root('tiertest', 'tier_test', 'dt');
SELECT tab_tier.bootstrap_tier_parts('tiertest', 'tier_test');

SELECT tab_tier.migrate_all_tiers();

SELECT count(*) FROM ONLY tier_test;

ROLLBACK;
