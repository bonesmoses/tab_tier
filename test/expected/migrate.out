\set ECHO none
CREATE EXTENSION tab_tier;
CREATE SCHEMA tiertest;
SET search_path TO tiertest;
CREATE TABLE tier_test (foo INT, dt TIMESTAMP WITH TIME ZONE);
INSERT INTO tier_test
SELECT a.id, '2016-12-15'::DATE - (a.id::TEXT || 'd')::INTERVAL
  FROM generate_series(1, 200) a (id);
SELECT tab_tier.register_tier_root('tiertest', 'tier_test', 'dt');
 register_tier_root 
--------------------
 
(1 row)

SELECT tab_tier.bootstrap_tier_parts('tiertest', 'tier_test');
 bootstrap_tier_parts 
----------------------
 
(1 row)

SELECT tab_tier.migrate_all_tiers();
 migrate_all_tiers 
-------------------
 
(1 row)

SELECT count(*) FROM ONLY tier_test;
 count 
-------
   187
(1 row)

ROLLBACK;
